src/internal/connector/onedrive/permission.go

```diff
@@ -89,6 +90,7 @@ func getCollectionMetadata(
 // Passing nil for the permissions results in just creating the folder(s).
 func createRestoreFoldersWithPermissions(
 	ctx context.Context,
+	creds account.M365Config,
```

Comment:
Having  to pass down `cdes` seems unncessary. Should we just add a method into `graph.Services` which gets us a new client from the available adapter?

Reply:
That was the original goal.  But the http client in the adapter is unexported.  We'd have to store the adapter's original http client (and, preferably, configuration) separately in the graph.Service.  It's an option, maybe one we'll return to, but this felt like a more direct solution in the meantime.